### ISSUE TITLE
Unable to rename Rmd file, shinyapps.io still uses old url

### ISSUE THREAD

**Issue by CBarraford** [rstudio/shinyapps#147]
After I deploy my rmd app, i want to rename the rmd file. After doing so, shinyapps package takes me to a url after deployment that is still the old url. 

###Recreation steps

1. deploy a new rmd shiny app to shinyapps.io. This should open your browser to a url similar to `account.shinyapps.io/rmd/mydoc.Rmd`
2. Rename your mydoc.Rmd file to something else and redeploy to shinyapps.io. When you goto `account.shinyapps.io/rmd` you get redirected to the old rmd file name (`account.shinyapps.io/rmd/mydoc.Rmd`)

**Comment by jjallaire** [comment]
I don't think you should bother with running this issue down. This is mostly because there are *many* enhancements to the way Rmd shiny apps are handled in the rsconnect and rmarkdown packages (and related changes to the IDE) including the ability to publish more than one Rmd file from a directory (e.g. ~/Desktop) and not have the deployment mechanism assume you want to publish the entire directory recursively!  There are also changes related to discovering actual and potential dependencies for shiny Rmd files and allowing users to whitelist files into the deployment.

Net: this issue doesn't occur in RStudio v0.99 / rsconnect because the package and IDE are smart about publishing just the right content (and giving users a voice in exactly what is published and how).

Besides the enhancements to Rmd publishing, rsconnect includes many enhancements related to linting including elimination of most classes of "false negative" and integration with the RStudio IDE source-marker API.

We should deprecate shinyapps and start promoting the rsconnect package to shinyapps customers (further explanation of why maintaining a separate shinyapps package is untenable is here: https://github.com/rstudio/shinyapps/pull/127#issuecomment-73232086). As noted in that comment, maintaining and promote shinyapps as a separate package has several problems:

1. You will get none of the enhancements that we've made to Rmd publishing and will have either punt these bugs in shinyapps or do additional work to resolve them.

2. You will get none of the other enhancements we make (e.g. to linting, to packrat-based dependency detection, etc.)

3. You will create confusion for customers as the IDE will *only* publish via rsconnect and will maintain it's own set of deployment records (it currently does a one-time migration of shinyapps deployment records).

I think the only sustainable alternative (that doesn't duplicate tons of code and effort) to a unified rsconnect package is to maintain 4 packages (the diamond dependency graph referenced in the comment above). We can do that, but it seems like a large amount of additional work for very little gain.
